Yoast/wpseo-news
# Use translationspress

## Context


* Since we are phasing out translate.yoast.com, we want to introduce the same mechanism which is in Premium and let users download the translations from Translationspress from inside their WP backend instead of bundling the .mo files

## Summary

This PR can be summarized in the following changelog entry:

* This PR introduces a new way of retrieving translations for Yoast News SEO, by utilizing the TranslationPress service. Instead of having to ship all translations with every release, we can now load the translations on a per-install basis, tailored to the user's setup. This means smaller plugin releases and less bloat on the user's server.

## Relevant technical choices:

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

**Note:** TranslatePress only holds some languages at the moment. However, they might not be fully translated. 

#### Prerequisites for all tests
* Navigate to Settings -> General and install at least one additional language if you don't have any already.
* Checkout this branch and build accordingly (`composer install`, `yarn`, `grunt build:dev`)
* Install and enable the Yoast Test helper: make sure to use 1.18-RC5 because it fixes an issue about the News SEO DB version update

#### Testing Grunt script
* Run `grunt build`.
* Check that the i18n related tasks run correctly and only the JSON files are pulled into the `languages` folder.

#### Testing manually downloading language files
* Delete the language files in `/wp-content/languages/plugins/` (or delete the entire `plugins` directory itself)
* Navigate to Dashboard -> Updates and scroll down to the the translations update button at the bottom of the page.
* Click the update button. See that the translation files for all installed plugins are being downloaded.
* Check the final output of this downloading process 
   * You should see a number of lines which don't specify the plugin but only the language. They should be the Video SEO 
ones, there is a problem since the language domain `wordpress-seo-news` does not match the folder `wpseo-news`. 
   * there should one such line for each installed language (except of course en_US)
* Navigate to Yoast SEO -> News SEO and see that the translations are working. Also test with additionally installed languages.
* Edit a post , check the `News` tab in the metabox and see the translations are working.

#### Testing the upgrade routine
* Delete the `wordpress-seo-news-*` language files in `/wp-content/languages/plugins/`.
* Navigate to Tools -> Yoast Test and set the version for News to 13.0 and click Save.
* Navigate to Yoast SEO -> News SEO  and see that the translations are working. Also test with additionally installed languages.

#### Testing outdated language files
* Navigate to `/wp-content/languages/plugins/` on your system.
* Editnews-seo-<preferred language code>.po` and change the `PO-Revision-Date` to be earlier than the current value (i.e. change it by one day or one month or one year).
* Save your changes.
* Navigate to Dashboard -> Updates and scroll down to the the translations update button at the bottom of the page.
* See that new translations are available. 
* Click the update button.
   * You should see a number of lines which don't specify the plugin but only the language. They should be the Woo SEO 
ones, there is a problem since the language domain `wordpress-seo-news` does not match the folder `wpseo-video`. 
   * there should just one such line for the language you edited.
* Navigate to Yoast SEO -> News SEO and see that the translations are working. Also test with additionally installed languages.
* Edit a post , check the `News` tab in the metabox and see the translations are working.


### Test instructions for QA when the code is in the RC


* [] QA should use the same steps as above.


QA can test this PR by following these steps:

* use the steps above except "Testing Grunt script"


## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Documentation


## Quality assurance


## Innovation


Fixes #
